frontend/src/Components/LoginPage/LoginPage.tsx

```diff
@@ -65,9 +59,23 @@ export const LoginPage = connectToWebSocket(({ socket }) => {
       <label for="session" class={classes.sessionLabel}>
         {LABEL_SESSION}
       </label>
-      <a id="session" href={sessionId && `?sessionId=${sessionId}`} class={classes.sessionLink}>
-        {sessionId}
-      </a>
+      <div class={classes.sessionContainer}>
+        <a id="session" href={sessionId && `?sessionId=${sessionId}`} class={classes.sessionLink}>
+          {sessionId}
+        </a>
+        <button
+          type="button"
+          class={classes.refreshButton}
+          onClick={() => setSessionId(refreshSessionId())}
+          title={BUTTON_REFRESH_SESSION}
+        >
+          {/* icon taken from material ui icons */}
+          <svg xmlns="http://www.w3.org/2000/svg" height="16px" viewBox="0 0 24 24" width="16px">
```

Comment:
Could we either extract the icon or the entire button (similar to the "KickButton") to a separate component for readability?

Reply:
Extracted the button.